src/OpenSleigh.Transport.RabbitMQ/PublisherChannelContextPool.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Concurrent;
+using RabbitMQ.Client;
+
+namespace OpenSleigh.Transport.RabbitMQ
+{
+    internal sealed class PublisherChannelContextPool : IPublisherChannelContextPool, IDisposable
+    {
+        private readonly IBusConnection _connection;
+        private readonly ConcurrentDictionary<string, ConcurrentBag<PublisherChannelContext>> _pools = new ();
```

Comment:
You can avoid stick the channel to a context, you can have a pool of generic channels, starting with some preallocated, create new ones as they need for concurrency and with a max number of it to avoid unbounded (max degree of parallelism, then more call will wait).

Reply:
I'll add this to another PR. thanks!